taskvine/src/worker/vine_cache.c

```diff
@@ -198,6 +199,52 @@ static int do_command( struct vine_cache *c, const char *command, const char *ca
 	return result;
 }
 
+/*
+Transfer a single input file from a worker url to a local file name. 
+
+*/
+static int do_worker_transfer( struct vine_cache *c, const char *source_url, const char *cache_path, char **error_message)
+{	
+	int port_num;
+	char *addr, *port, *path;
+	char source_mutable[4096];
+	
+	strcpy(source_mutable, source_url);
+	
+	// expect the form: worker://addr:port/path/to/file
```

Comment:
You can use `sscanf` to parse all this stuff out in one go.
Note that the format specifiers can be used sort of like a regex.